### ISSUE TITLE
Double Call for viewWillAppear and viewDidAppear 

### ISSUE THREAD

**Issue by Lotiq** [jonkykong/SideMenu#473]
**I have read the [guidelines for contributing](https://github.com/jonkykong/SideMenu/blob/master/.github/CONTRIBUTING.md) and I understand**
- [x] My issue is happening in the **latest version** of SideMenu (older versions are no longer maintained).
- [x] My issue was **not** solved in the [README](https://github.com/jonkykong/SideMenu/blob/master/README.md).
- [x] My issue can **not** be answered on [stackoverflow.com](stackoverflow.com).
- [x] My issue is **not** a request for new functionality that I am unwilling to build and contribute with a pull request.
- [x] My issue **is** reproducible in the [demo project](https://github.com/jonkykong/SideMenu/tree/master/Example).

**Describe the bug**
A side menu is presented and contains a button, which in turn, when pressed, pushes another controller. When that other controller is presented, viewWillAppear and viewWillDisappear gets called twice

**To Reproduce**
1. Create initial VC (VC1) as a root vc for a navigation controller.
2. Add side menu that contains a button within.
3. Add another VC (VC2) that gets presented when the button is pressed.
4. When VC2 is presented viewWillAppear and viewWillDisappear get called twice

**Expected behavior**
Only single call on viewWillAppear and viewWillDisappear

**Comment by EndLess728** [comment]
same issue here

**Comment by fplsv** [comment]
same

I guess this is related

`Unbalanced calls to begin/end appearance transitions for <project.Controller:0x123131231>`

**Comment by vininandu** [comment]
I too got same issue ,I have done work around for this issue
        DispatchQueue.main.async {
            self.dismiss(animated: false, completion: nil)
        }

**Comment by EndLess728** [comment]
Got it Fixed By Doing this 

```swift
      //do this in viewDidLoad 
        let FirstViewController = storyboard!.instantiateViewController(withIdentifier: "MenuViewController") as? MenuViewController //dont call UISidemenuNavigation here 
        let menuRightNavigationController = UISideMenuNavigationController(rootViewController: FirstViewController!)
        SideMenuManager.default.menuRightNavigationController = menuRightNavigationController
        SideMenuManager.default.menuAddScreenEdgePanGesturesToPresent(toView: (self.navigationController?.view)!)
        SideMenuManager.default.menuPresentMode = .viewSlideOut //.menuSlideIn
        SideMenuManager.default.menuWidth = UIScreen.main.bounds.size.width - 150
        SideMenuManager.default.menuFadeStatusBar = true
        SideMenuManager.default.menuAnimationBackgroundColor  = UIColor.clear
        SideMenuManager.default.menuShadowColor = UIColor.lightGray
        SideMenuManager.default.menuShadowOpacity = 0.3
        SideMenuManager.default.menuShadowRadius = 3

    //Now Call it Like this 
    @IBAction func MenuButtonAction(_ sender: Any) {
       / /dont call UISidemenuNavigation here 
        self.present(SideMenuManager.default.menuRightNavigationController!, animated: true, completion: nil)

    }
```

**Comment by jonkykong** [comment]
This issue should now be resolved in version 6.0.0+ of SideMenu. Please run `pod update` and reopen this issue if the problem persists.
